nativescript-angular/view-util.ts

```diff
@@ -249,8 +264,11 @@ export class ViewUtil {
             this.removeLayoutChild(parent, child);
         } else if (isContentView(parent) && parent.content === child) {
             parent.content = null;
-            parent.lastChild = null;
-            parent.firstChild = null;
+        } else if (child.nodeName === "DetachedContainer") {
```

Comment:
I would suggest more general approach. Let's use the `isDetachedElement` to restrict calling the [`removeFromVisualTree` at all](https://github.com/NativeScript/nativescript-angular/blob/2a64e87207c86bfa14533daaccea9d46ea007e1b/nativescript-angular/view-util.ts#L185) (this will work for all types of elements that are not added to the visual tree):
```
if (!isDetachedElement(extendedChild)) {
    this.removeFromVisualTree(extendedParent, extendedChild);
}
```

There is the same logic when [calling `addToVisualTree`](https://github.com/NativeScript/nativescript-angular/blob/2a64e87207c86bfa14533daaccea9d46ea007e1b/nativescript-angular/view-util.ts#L75-L78).

Reply:
Much better thank you.
I've made the suggested change.